File: backend/routes/export.py
"""
Export Routes
=============

API endpoints for exporting prediction results to CSV and PDF formats.

Author: EcoPackAI Team
Date: 2026-01-12
"""

from flask import Blueprint, jsonify, request, send_file, make_response
from datetime import datetime
import pandas as pd
import io
import logging
from typing import Dict, Any

logger = logging.getLogger(__name__)

# Create blueprint
export_bp = Blueprint('export', __name__, url_prefix='/api/v1/export')


def generate_csv_from_prediction(prediction_data: Dict) -> str:
    """
    Generate CSV content from prediction data
    
    Args:
        prediction_data: Prediction result dictionary
        
    Returns:
        CSV string content
    """
    rows = []
    
    # Add metadata
    rows.append(['Parameter', 'Value'])
    rows.append(['Timestamp', prediction_data.get('timestamp', datetime.now().isoformat())])
    rows.append(['Prediction Type', prediction_data.get('prediction_type', 'all')])
    rows.append(['', ''])  # Empty row
    
    # Add results
    if 'results' in prediction_data:
        rows.append(['RESULTS', ''])
        for key, value in prediction_data['results'].items():
            param_name = key.replace('_', ' ').title()
            formatted_value = f"{value:.4f}" if isinstance(value, float) else str(value)
            rows.append([param_name, formatted_value])
        rows.append(['', ''])
    
    # Add metadata
    if 'metadata' in prediction_data:
        rows.append(['METADATA', ''])
        for key, value in prediction_data['metadata'].items():
            param_name = key.replace('_', ' ').title()
            rows.append([param_name, str(value)])
    
    # Convert to CSV
    df = pd.DataFrame(rows)
    csv_content = df.to_csv(index=False, header=False)
    
    return csv_content


def generate_materials_comparison_csv(materials: list) -> str:
    """
    Generate CSV for materials comparison
    
    Args:
        materials: List of material dictionaries
        
    Returns:
        CSV string content
    """
    export_data = []
    
    for idx, material in enumerate(materials):
        row = {
            'Material ID': material.get('id') or material.get('name') or f'Material {idx + 1}',
            'Predicted Cost ($)': f"{material.get('predicted_cost', 0):.2f}",
            'Predicted CO₂ (kg)': f"{material.get('predicted_co2', 0):.4f}",
            'Cost Confidence': f"{material.get('cost_confidence', 0):.4f}",
            'Recyclability (%)': material.get('recyclability_percent', 'N/A'),
            'Recycled Content (%)': material.get('recycled_content_percent', 'N/A'),
            'Sustainability Score': f"{material.get('overall_sustainability_score', 0):.4f}",
            'Material Suitability': material.get('material_suitability_score', 'N/A')
        }
        export_data.append(row)
    
    df = pd.DataFrame(export_data)
    csv_content = df.to_csv(index=False)
    
    return csv_content


@export_bp.route('/csv', methods=['POST'])
def export_csv():
    """
    Export prediction results to CSV
    
    Request Body:
        {
            "prediction_data": {...},
            "export_type": "single" | "comparison",
            "materials": [...] (for comparison export)
        }
    
    Response:
        200: CSV file download
        400: Invalid input
        500: Export error
    """
    try:
        data = request.get_json()
        
        if not data:
            return jsonify({
                "status": "error",
                "timestamp": datetime.now().isoformat(),
                "error": "No data provided"
            }), 400
        
        export_type = data.get('export_type', 'single')
        
        if export_type == 'single':
            # Export single prediction
            prediction_data = data.get('prediction_data')
            if not prediction_data:
                return jsonify({
                    "status": "error",
                    "timestamp": datetime.now().isoformat(),
                    "error": "Missing prediction_data"
                }), 400
            
            csv_content = generate_csv_from_prediction(prediction_data)
            filename = f"ecopack_prediction_{datetime.now().strftime('%Y%m%d_%H%M%S')}.csv"
            
        elif export_type == 'comparison':
            # Export materials comparison
            materials = data.get('materials')
            if not materials or not isinstance(materials, list):
                return jsonify({
                    "status": "error",
                    "timestamp": datetime.now().isoformat(),
                    "error": "Missing or invalid materials array"
                }), 400
            
            csv_content = generate_materials_comparison_csv(materials)
            filename = f"materials_comparison_{datetime.now().strftime('%Y%m%d_%H%M%S')}.csv"
        
        else:
            return jsonify({
                "status": "error",
                "timestamp": datetime.now().isoformat(),
                "error": f"Unknown export_type: {export_type}"
            }), 400
        
        # Create response with CSV file
        output = io.StringIO()
        output.write(csv_content)
        output.seek(0)
        
        response = make_response(output.getvalue())
        response.headers['Content-Type'] = 'text/csv'
        response.headers['Content-Disposition'] = f'attachment; filename={filename}'
        
        logger.info(f"CSV export successful: {filename}")
        return response
        
    except Exception as e:
        logger.error(f"CSV export error: {e}")
        return jsonify({
            "status": "error",
            "timestamp": datetime.now().isoformat(),
            "error": "Export Error",
            "message": str(e)
        }), 500


@export_bp.route('/pdf', methods=['POST'])
def export_pdf():
    """
    Export prediction results to PDF
    
    Note: This is a placeholder endpoint. PDF generation requires
    additional libraries like ReportLab or WeasyPrint.
    
    Request Body:
        {
            "prediction_data": {...},
            "include_charts": true|false
        }
    
    Response:
        200: PDF file download
        400: Invalid input
        501: Not implemented (if PDF library not available)
    """
    try:
        data = request.get_json()
        
        if not data:
            return jsonify({
                "status": "error",
                "timestamp": datetime.now().isoformat(),
                "error": "No data provided"
            }), 400
        
        # Check if PDF libraries are available
        try:
            from reportlab.lib.pagesizes import letter
            from reportlab.lib import colors
            from reportlab.lib.styles import getSampleStyleSheet, ParagraphStyle
            from reportlab.platypus import SimpleDocTemplate, Table, TableStyle, Paragraph, Spacer
            from reportlab.lib.units import inch
        except ImportError:
            logger.warning("ReportLab not installed, PDF export unavailable")
            return jsonify({
                "status": "error",
                "timestamp": datetime.now().isoformat(),
                "error": "PDF Export Not Available",
                "message": "PDF generation library not installed. Please use client-side PDF export."
            }), 501
        
        prediction_data = data.get('prediction_data')
        if not prediction_data:
            return jsonify({
                "status": "error",
                "timestamp": datetime.now().isoformat(),
                "error": "Missing prediction_data"
            }), 400
        
        # Generate PDF
        buffer = io.BytesIO()
        doc = SimpleDocTemplate(buffer, pagesize=letter)
        elements = []
        styles = getSampleStyleSheet()
        
        # Title
        title_style = ParagraphStyle(
            'CustomTitle',
            parent=styles['Heading1'],
            fontSize=24,
            textColor=colors.HexColor('#10B981'),
            spaceAfter=30
        )
        title = Paragraph("EcoPackAI Sustainability Report", title_style)
        elements.append(title)
        
        # Subtitle
        subtitle = Paragraph(
            "AI-Powered Packaging Analysis",
            styles['Heading2']
        )
        elements.append(subtitle)
        elements.append(Spacer(1, 0.3*inch))
        
        # Metadata
        metadata_data = [
            ['Report Date:', datetime.now().strftime('%Y-%m-%d %H:%M:%S')],
            ['Prediction Time:', prediction_data.get('timestamp', 'N/A')]
        ]
        
        metadata_table = Table(metadata_data, colWidths=[2*inch, 4*inch])
        metadata_table.setStyle(TableStyle([
            ('FONT', (0, 0), (-1, -1), 'Helvetica', 10),
            ('FONT', (0, 0), (0, -1), 'Helvetica-Bold', 10),
            ('ALIGN', (0, 0), (-1, -1), 'LEFT'),
            ('VALIGN', (0, 0), (-1, -1), 'MIDDLE'),
            ('GRID', (0, 0), (-1, -1), 0.5, colors.grey),
        ]))
        elements.append(metadata_table)
        elements.append(Spacer(1, 0.3*inch))
        
        # Results
        if 'results' in prediction_data:
            results_title = Paragraph("Prediction Results", styles['Heading2'])
            elements.append(results_title)
            elements.append(Spacer(1, 0.2*inch))
            
            results = prediction_data['results']
            results_data = []
            
            for key, value in results.items():
                param_name = key.replace('_', ' ').title()
                formatted_value = f"{value:.4f}" if isinstance(value, float) else str(value)
                results_data.append([param_name, formatted_value])
            
            results_table = Table(results_data, colWidths=[3*inch, 3*inch])
            results_table.setStyle(TableStyle([
                ('FONT', (0, 0), (-1, -1), 'Helvetica', 10),
                ('FONT', (0, 0), (0, -1), 'Helvetica-Bold', 10),
                ('ALIGN', (0, 0), (-1, -1), 'LEFT'),
                ('VALIGN', (0, 0), (-1, -1), 'MIDDLE'),
                ('GRID', (0, 0), (-1, -1), 0.5, colors.grey),
                ('BACKGROUND', (0, 0), (0, -1), colors.HexColor('#F3F4F6')),
            ]))
            elements.append(results_table)
            elements.append(Spacer(1, 0.3*inch))
        
        # Model Information
        if 'metadata' in prediction_data:
            model_title = Paragraph("Model Information", styles['Heading2'])
            elements.append(model_title)
            elements.append(Spacer(1, 0.2*inch))
            
            metadata = prediction_data['metadata']
            model_data = []
            
            for key, value in metadata.items():
                param_name = key.replace('_', ' ').title()
                model_data.append([param_name, str(value)])
            
            model_table = Table(model_data, colWidths=[3*inch, 3*inch])
            model_table.setStyle(TableStyle([
                ('FONT', (0, 0), (-1, -1), 'Helvetica', 10),
                ('FONT', (0, 0), (0, -1), 'Helvetica-Bold', 10),
                ('ALIGN', (0, 0), (-1, -1), 'LEFT'),
                ('VALIGN', (0, 0), (-1, -1), 'MIDDLE'),
                ('GRID', (0, 0), (-1, -1), 0.5, colors.grey),
                ('BACKGROUND', (0, 0), (0, -1), colors.HexColor('#F3F4F6')),
            ]))
            elements.append(model_table)
        
        # Build PDF
        doc.build(elements)
        
        buffer.seek(0)
        filename = f"ecopack_report_{datetime.now().strftime('%Y%m%d_%H%M%S')}.pdf"
        
        return send_file(
            buffer,
            mimetype='application/pdf',
            as_attachment=True,
            download_name=filename
        )
        
    except Exception as e:
        logger.error(f"PDF export error: {e}")
        return jsonify({
            "status": "error",
            "timestamp": datetime.now().isoformat(),
            "error": "Export Error",
            "message": str(e)
        }), 500


@export_bp.route('/formats', methods=['GET'])
def get_export_formats():
    """
    Get available export formats
    
    Response:
        200: List of available formats
    """
    formats = {
        "csv": {
            "name": "CSV (Comma-Separated Values)",
            "extension": ".csv",
            "mime_type": "text/csv",
            "available": True,
            "description": "Export data in CSV format for Excel and data analysis tools"
        },
        "pdf": {
            "name": "PDF (Portable Document Format)",
            "extension": ".pdf",
            "mime_type": "application/pdf",
            "available": False,  # Will be set based on library availability
            "description": "Export formatted report as PDF document"
        }
    }
    
    # Check PDF availability
    try:
        import reportlab
        formats["pdf"]["available"] = True
    except ImportError:
        pass
    
    return jsonify({
        "status": "success",
        "timestamp": datetime.now().isoformat(),
        "formats": formats
    }), 200

File: frontend/static/js/export.js
/**
 * EcoPackAI Export Functionality
 * ================================
 * 
 * Export prediction results to CSV and PDF formats
 * Supports both frontend download and backend generation
 * 
 * Author: EcoPackAI Team
 * Date: 2026-01-12
 */

class ExportManager {
    constructor() {
        this.apiBase = '/api/v1';
    }

    /**
     * Export data to CSV format
     * @param {Array} data - Array of prediction objects
     * @param {string} filename - Output filename
     */
    exportToCSV(data, filename = 'ecopack_predictions.csv') {
        if (!data || data.length === 0) {
            console.error('No data to export');
            alert('No data available for export');
            return;
        }

        try {
            // Extract headers from first object
            const headers = Object.keys(data[0]);

            // Create CSV content
            let csvContent = '';

            // Add headers
            csvContent += headers.map(h => this.escapeCSVValue(h)).join(',') + '\n';

            // Add data rows
            data.forEach(row => {
                const values = headers.map(header => {
                    const value = row[header];
                    return this.escapeCSVValue(value);
                });
                csvContent += values.join(',') + '\n';
            });

            // Create blob and download
            this.downloadFile(csvContent, filename, 'text/csv;charset=utf-8;');

            console.log(`Exported ${data.length} rows to ${filename}`);
            return true;
        } catch (error) {
            console.error('CSV export error:', error);
            alert('Failed to export CSV: ' + error.message);
            return false;
        }
    }

    /**
     * Escape CSV values to handle commas, quotes, and newlines
     * @param {*} value - Value to escape
     * @returns {string} Escaped value
     */
    escapeCSVValue(value) {
        if (value === null || value === undefined) {
            return '';
        }

        const stringValue = String(value);

        // If value contains comma, quote, or newline, wrap in quotes
        if (stringValue.includes(',') || stringValue.includes('"') || stringValue.includes('\n')) {
            return `"${stringValue.replace(/"/g, '""')}"`;
        }

        return stringValue;
    }

    /**
     * Export prediction results with detailed metadata
     * @param {Object} predictionResult - Complete prediction result object
     * @param {string} filename - Output filename
     */
    exportPredictionToCSV(predictionResult, filename = 'prediction_results.csv') {
        if (!predictionResult) {
            console.error('No prediction result to export');
            return;
        }

        try {
            const data = [];

            // Add metadata
            data.push({
                'Parameter': 'Timestamp',
                'Value': predictionResult.timestamp || new Date().toISOString()
            });

            data.push({
                'Parameter': 'Prediction Type',
                'Value': predictionResult.prediction_type || 'all'
            });

            data.push({ 'Parameter': '', 'Value': '' }); // Empty row

            // Add results
            if (predictionResult.results) {
                data.push({ 'Parameter': 'RESULTS', 'Value': '' });

                Object.entries(predictionResult.results).forEach(([key, value]) => {
                    data.push({
                        'Parameter': key.replace(/_/g, ' ').replace(/\b\w/g, l => l.toUpperCase()),
                        'Value': typeof value === 'number' ? value.toFixed(4) : value
                    });
                });
            }

            // Add metadata if present
            if (predictionResult.metadata) {
                data.push({ 'Parameter': '', 'Value': '' }); // Empty row
                data.push({ 'Parameter': 'METADATA', 'Value': '' });

                Object.entries(predictionResult.metadata).forEach(([key, value]) => {
                    data.push({
                        'Parameter': key.replace(/_/g, ' ').replace(/\b\w/g, l => l.toUpperCase()),
                        'Value': value
                    });
                });
            }

            this.exportToCSV(data, filename);
        } catch (error) {
            console.error('Prediction export error:', error);
            alert('Failed to export prediction: ' + error.message);
        }
    }

    /**
     * Export multiple materials comparison to CSV
     * @param {Array} materials - Array of material predictions
     * @param {string} filename - Output filename
     */
    exportMaterialsComparison(materials, filename = 'materials_comparison.csv') {
        if (!materials || materials.length === 0) {
            alert('No materials to export');
            return;
        }

        try {
            const exportData = materials.map((material, index) => ({
                'Material ID': material.id || material.name || `Material ${index + 1}`,
                'Predicted Cost ($)': material.predicted_cost?.toFixed(2) || 'N/A',
                'Predicted CO₂ (kg)': material.predicted_co2?.toFixed(4) || 'N/A',
                'Cost Confidence': material.cost_confidence?.toFixed(4) || 'N/A',
                'Recyclability (%)': material.recyclability_percent || 'N/A',
                'Recycled Content (%)': material.recycled_content_percent || 'N/A',
                'Sustainability Score': material.overall_sustainability_score?.toFixed(4) || 'N/A',
                'Material Suitability': material.material_suitability_score || 'N/A'
            }));

            this.exportToCSV(exportData, filename);
        } catch (error) {
            console.error('Materials comparison export error:', error);
            alert('Failed to export materials comparison: ' + error.message);
        }
    }

    /**
     * Generate PDF report (client-side using jsPDF)
     * @param {Object} predictionResult - Prediction result object
     * @param {string} filename - Output filename
     */
    async exportToPDF(predictionResult, filename = 'ecopack_report.pdf') {
        // Check if jsPDF is available
        if (typeof window.jsPDF === 'undefined' && typeof window.jspdf === 'undefined') {
            console.error('jsPDF library not loaded');
            alert('PDF export library not available. Please ensure jsPDF is included.');
            return;
        }

        try {
            // Support both window.jsPDF and window.jspdf for compatibility
            const jsPDF = window.jsPDF || window.jspdf?.jsPDF;
            if (!jsPDF) {
                throw new Error('jsPDF constructor not found');
            }
            const doc = new jsPDF();

            // Set up document
            let yPosition = 20;
            const lineHeight = 10;
            const pageWidth = doc.internal.pageSize.getWidth();
            const margin = 20;

            // Title
            doc.setFontSize(20);
            doc.setFont('helvetica', 'bold');
            doc.text('EcoPackAI Sustainability Report', margin, yPosition);

            yPosition += lineHeight * 1.5;

            // Subtitle
            doc.setFontSize(12);
            doc.setFont('helvetica', 'normal');
            doc.setTextColor(100);
            doc.text('AI-Powered Packaging Analysis', margin, yPosition);

            yPosition += lineHeight * 2;

            // Divider line
            doc.setDrawColor(200);
            doc.line(margin, yPosition, pageWidth - margin, yPosition);
            yPosition += lineHeight;

            // Metadata
            doc.setFontSize(10);
            doc.setTextColor(0);
            doc.setFont('helvetica', 'bold');
            doc.text('Report Date:', margin, yPosition);
            doc.setFont('helvetica', 'normal');
            doc.text(new Date().toLocaleString(), margin + 50, yPosition);
            yPosition += lineHeight;

            if (predictionResult.timestamp) {
                doc.setFont('helvetica', 'bold');
                doc.text('Prediction Time:', margin, yPosition);
                doc.setFont('helvetica', 'normal');
                doc.text(new Date(predictionResult.timestamp).toLocaleString(), margin + 50, yPosition);
                yPosition += lineHeight;
            }

            yPosition += lineHeight;

            // Results Section
            if (predictionResult.results) {
                doc.setFontSize(14);
                doc.setFont('helvetica', 'bold');
                doc.text('Prediction Results', margin, yPosition);
                yPosition += lineHeight * 1.5;

                doc.setFontSize(10);
                const results = predictionResult.results;

                // Cost
                if (results.predicted_cost !== undefined) {
                    doc.setFont('helvetica', 'bold');
                    doc.text('Predicted Cost:', margin + 5, yPosition);
                    doc.setFont('helvetica', 'normal');
                    doc.setTextColor(16, 185, 129); // Green
                    doc.text(`$${results.predicted_cost.toFixed(2)}`, margin + 60, yPosition);
                    doc.setTextColor(0);
                    yPosition += lineHeight;
                }

                // Cost Confidence
                if (results.cost_confidence !== undefined) {
                    doc.setFont('helvetica', 'bold');
                    doc.text('Cost Confidence:', margin + 5, yPosition);
                    doc.setFont('helvetica', 'normal');
                    doc.text(`${(results.cost_confidence * 100).toFixed(2)}%`, margin + 60, yPosition);
                    yPosition += lineHeight;
                }

                yPosition += 5;

                // CO2
                if (results.predicted_co2 !== undefined) {
                    doc.setFont('helvetica', 'bold');
                    doc.text('Predicted CO₂:', margin + 5, yPosition);
                    doc.setFont('helvetica', 'normal');
                    doc.setTextColor(59, 130, 246); // Blue
                    doc.text(`${results.predicted_co2.toFixed(4)} kg`, margin + 60, yPosition);
                    doc.setTextColor(0);
                    yPosition += lineHeight;
                }

                yPosition += lineHeight;
            }

            // Model Information Section
            if (predictionResult.metadata) {
                doc.setFontSize(14);
                doc.setFont('helvetica', 'bold');
                doc.text('Model Information', margin, yPosition);
                yPosition += lineHeight * 1.5;

                doc.setFontSize(10);
                const metadata = predictionResult.metadata;

                if (metadata.cost_model) {
                    doc.setFont('helvetica', 'bold');
                    doc.text('Cost Model:', margin + 5, yPosition);
                    doc.setFont('helvetica', 'normal');
                    doc.text(metadata.cost_model, margin + 50, yPosition);
                    yPosition += lineHeight;
                }

                if (metadata.co2_model) {
                    doc.setFont('helvetica', 'bold');
                    doc.text('CO₂ Model:', margin + 5, yPosition);
                    doc.setFont('helvetica', 'normal');
                    doc.text(metadata.co2_model, margin + 50, yPosition);
                    yPosition += lineHeight;
                }

                if (metadata.cost_r2) {
                    doc.setFont('helvetica', 'bold');
                    doc.text('Cost Model R²:', margin + 5, yPosition);
                    doc.setFont('helvetica', 'normal');
                    doc.text((metadata.cost_r2 * 100).toFixed(2) + '%', margin + 50, yPosition);
                    yPosition += lineHeight;
                }

                if (metadata.co2_r2) {
                    doc.setFont('helvetica', 'bold');
                    doc.text('CO₂ Model R²:', margin + 5, yPosition);
                    doc.setFont('helvetica', 'normal');
                    doc.text((metadata.co2_r2 * 100).toFixed(2) + '%', margin + 50, yPosition);
                    yPosition += lineHeight;
                }
            }

            // Footer
            const footerY = doc.internal.pageSize.getHeight() - 20;
            doc.setFontSize(8);
            doc.setTextColor(150);
            doc.text('Generated by EcoPackAI - AI-Powered Sustainable Packaging System', margin, footerY);
            doc.text('© 2026 EcoPackAI. All rights reserved.', margin, footerY + 5);

            // Save PDF
            doc.save(filename);

            console.log(`PDF exported: ${filename}`);
            return true;
        } catch (error) {
            console.error('PDF export error:', error);
            alert('Failed to export PDF: ' + error.message);
            return false;
        }
    }

    /**
     * Request server-side PDF generation
     * @param {Object} predictionData - Data to include in PDF
     * @param {string} filename - Desired filename
     */
    async requestServerPDF(predictionData, filename = 'ecopack_report.pdf') {
        try {
            const response = await fetch(`${this.apiBase}/export/pdf`, {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                },
                body: JSON.stringify(predictionData)
            });

            if (!response.ok) {
                throw new Error(`Server responded with ${response.status}`);
            }

            // Download the PDF
            const blob = await response.blob();
            const url = window.URL.createObjectURL(blob);
            const a = document.createElement('a');
            a.href = url;
            a.download = filename;
            document.body.appendChild(a);
            a.click();
            window.URL.revokeObjectURL(url);
            document.body.removeChild(a);

            console.log('Server-generated PDF downloaded');
            return true;
        } catch (error) {
            console.error('Server PDF export error:', error);
            // Fallback to client-side generation
            console.log('Falling back to client-side PDF generation');
            return this.exportToPDF(predictionData, filename);
        }
    }

    /**
     * Download file helper
     * @param {string} content - File content
     * @param {string} filename - Filename
     * @param {string} mimeType - MIME type
     */
    downloadFile(content, filename, mimeType) {
        const blob = new Blob([content], { type: mimeType });
        const url = window.URL.createObjectURL(blob);
        const a = document.createElement('a');
        a.href = url;
        a.download = filename;
        document.body.appendChild(a);
        a.click();
        window.URL.revokeObjectURL(url);
        document.body.removeChild(a);
    }

    /**
     * Export current page data (for use in results pages)
     * @param {string} format - 'csv' or 'pdf'
     */
    exportCurrentData(format = 'csv') {
        // Try to get data from localStorage or sessionStorage
        const storedData = localStorage.getItem('latestPrediction') ||
            sessionStorage.getItem('latestPrediction');

        if (!storedData) {
            alert('No prediction data available to export');
            return;
        }

        try {
            const data = JSON.parse(storedData);

            if (format === 'csv') {
                this.exportPredictionToCSV(data);
            } else if (format === 'pdf') {
                this.exportToPDF(data);
            } else {
                console.error('Unsupported export format:', format);
            }
        } catch (error) {
            console.error('Export error:', error);
            alert('Failed to export data: ' + error.message);
        }
    }
}

// Export singleton instance
const exportManager = new ExportManager();

// Make available globally
window.ExportManager = exportManager;

// Add export buttons event listeners when DOM is ready
document.addEventListener('DOMContentLoaded', () => {
    // CSV export button
    const csvButton = document.getElementById('exportCSV');
    if (csvButton) {
        csvButton.addEventListener('click', () => {
            exportManager.exportCurrentData('csv');
        });
    }

    // PDF export button
    const pdfButton = document.getElementById('exportPDF');
    if (pdfButton) {
        pdfButton.addEventListener('click', () => {
            exportManager.exportCurrentData('pdf');
        });
    }
});
